### PR TITLE
[MM-61761] Fix errcheck issues in channels/app/platform/cluster_handlers.go

### PR DESCRIPTION
**Pull request by Honsei901**
#### Summary
Fix errcheck issues in channels/app/platform/cluster_handlers.go

#### Ticket Link
Fixes https://github.com/mattermost/mattermost/issues/29263
Jira MM-61761

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by hanzei — APPROVED** [review]
LGTM :+1:

File: server/channels/app/platform/cluster_handlers.go [hunk 1]
```diff
@@ -60,7 +60,9 @@ func (ps *PlatformService) ClusterUpdateStatusHandler(msg *model.ClusterMessage)
 		ps.logger.Warn("Failed to decode status from JSON")
 	}
 
-	ps.statusCache.SetWithDefaultExpiry(status.UserId, status)
+	if err := ps.statusCache.SetWithDefaultExpiry(status.UserId, status); err != nil {
+		ps.logger.Warn("Failed to store the given key and value")
```
Comment (hanzei) [inline server/channels/app/platform/cluster_handlers.go:64]: Let's also log the error here and in the rest of the file using `mlog.Err`.
```suggestion
		ps.logger.Warn("Failed to store the given key and value", mlog.Err(err))
```

**Review by hanzei — CHANGES_REQUESTED** [review]
Whoops, missed something

File: server/channels/app/platform/cluster_handlers.go (hunk 1 above)
Comment (Honsei901) [reply server/channels/app/platform/cluster_handlers.go]: I've resolved the issue.

**Review by hanzei — APPROVED** [review]
LGTM :+1:

File: server/channels/app/platform/cluster_handlers.go [hunk 2]
```diff
@@ -57,10 +57,12 @@ func (ps *PlatformService) ClusterPublishHandler(msg *model.ClusterMessage) {
 func (ps *PlatformService) ClusterUpdateStatusHandler(msg *model.ClusterMessage) {
 	var status model.Status
 	if jsonErr := json.Unmarshal(msg.Data, &status); jsonErr != nil {
-		ps.logger.Warn("Failed to decode status from JSON")
+		ps.logger.Warn("Failed to decode status from JSON", mlog.Err(jsonErr))
 	}
 
-	ps.statusCache.SetWithDefaultExpiry(status.UserId, status)
+	if err := ps.statusCache.SetWithDefaultExpiry(status.UserId, status); err != nil {
+		ps.logger.Warn("Failed to store the given key and value", mlog.Err(err))
```
Comment (agnivade) [inline server/channels/app/platform/cluster_handlers.go:64]:
```suggestion
		ps.logger.Warn("Failed to store the status in the cache", mlog.String("user_id", status.UserId), mlog.Err(err))
```

**Review by agnivade — CHANGES_REQUESTED** [review]

**Comment by agnivade** [comment]
You'll also have to resolve the conflicts.

**Comment by Honsei901** [comment]
I've resolved the issues.

File: server/channels/app/platform/cluster_handlers.go [hunk 3]
```diff
@@ -91,8 +93,8 @@ func (ps *PlatformService) clusterClearSessionCacheForAllUsersHandler(msg *model
 
 func (ps *PlatformService) clusterBusyStateChgHandler(msg *model.ClusterMessage) {
 	var sbs model.ServerBusyState
-	if jsonErr := json.Unmarshal(msg.Data, &sbs); jsonErr != nil {
-		mlog.Warn("Failed to decode server busy state from JSON", mlog.Err(jsonErr))
+	if err := json.Unmarshal(msg.Data, &sbs); err != nil {
+		mlog.Warn("Failed to decode server busy state from JSON", mlog.Err(err))
```
Comment (agnivade) [inline server/channels/app/platform/cluster_handlers.go:97]:
```suggestion
		ps.logger.Warn("Failed to decode server busy state from JSON", mlog.Err(err))
```

**Review by agnivade — CHANGES_REQUESTED** [review]

File: server/channels/app/platform/cluster_handlers.go (hunk 3 above)
Comment (Honsei901) [reply server/channels/app/platform/cluster_handlers.go]: I've resolved.

**Review by agnivade — APPROVED** [review]

**Pull request merged**